packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -327,5 +329,13 @@ function useMissingTemplates(
 		...postTypesMenuItems,
 		...taxonomiesMenuItems,
 	];
-	return missingTemplates;
+
+	const missingTemplateSlugs = ( missingTemplates || [] ).map(
+		( { slug } ) => slug
+	);
+	const missingExtraTemplateTypes = ( extraTemplateTypes || [] ).filter(
```

Comment:
We should also check if the extra template already exists(`existingTemplateSlugs`) and make sure that no override of the `default` templates can be injected, as we handle all these automatically. Related to this we should mention in the `extra_template_types` filter docs the need for it and what is filtered out. 

Finally we need to add to the extra templates an `onClick={ (template) => createTemplate(template, false)` because we have the `isWPSuggestion` flag we need to set to `false`.